src/contracts/account/ReplayProtection.sol

```diff
@@ -71,38 +86,51 @@ contract ReplayProtection {
      * @param _replayProtection Contains a single nonce
      */
     function nonce(address _signer, bytes memory _replayProtection) internal returns(bool) {
-        uint256 nonce1;
-        uint256 nonce2;
+        uint256 queue;
+        uint256 queueNonce;
 
-        (nonce1, nonce2) = abi.decode(_replayProtection, (uint256, uint256));
-        bytes32 index = keccak256(abi.encode(_signer, nonce1));
+        (queue, queueNonce) = abi.decode(_replayProtection, (uint256, uint256));
+        bytes32 index = queueIndex(_signer, queue, getMultiNonceAddress());
         uint256 storedNonce = nonceStore[index];
 
         // Increment stored nonce by one...
-        if(nonce2 == storedNonce) {
+        if(queueNonce == storedNonce) {
             nonceStore[index] = storedNonce + 1;
             return true;
         }
 
         return false;
     }
-
     /**
      * Bitflip Replay Protection
      * Explained: https://github.com/PISAresearch/metamask-comp#bitflip
      * Allows a user to flip a bit in nonce2 as replay protection. Every nonce supports 256 bit flips.
      */
     function bitflip(address _signer, bytes memory _replayProtection) internal returns(bool) {
-        (uint256 nonce1, uint256 bitsToFlip) = abi.decode(_replayProtection, (uint256, uint256));
+        (uint256 queue, uint256 bitsToFlip) = abi.decode(_replayProtection, (uint256, uint256));
 
-        // It is unlikely that anyone will need to send 6174 concurrent transactions per block,
-        // plus 6174 is a cool af number.
-        require(nonce1 >= 6174, "Nonce1 must be at least 6174 to separate multinonce and bitflip");
+        require(bitsToFlip > 0, "It must flip at least one bit!");
 
         // Combine with msg.sender to get unique indexes per caller
-        bytes32 senderIndex = keccak256(abi.encodePacked(_signer, nonce1));
-        uint256 currentBitmap = nonceStore[senderIndex];
-        require(currentBitmap & bitsToFlip != bitsToFlip, "Bit already flipped.");
-        nonceStore[senderIndex] = currentBitmap | bitsToFlip;
+        bytes32 index = queueIndex(_signer, queue, getBitFlipAddress());
+        uint256 currentBitmap = nonceStore[index];
+
+        // Finally check if the bit is flipped!
```

Comment:
comment should say "check if at least one bit is flipped"

Reply:
fixed